### PR TITLE
Add devise

### PR DESCRIPTION
**Pull request by rbanos-mv**
# Ruby Blog app

> The Blog app will be a classic example of a blog website. You will create a fully functional website that will show the list of posts and empower readers to interact with them by adding comments and liking posts.

### Project requirements
- [x] Remove `current_user` method in `ApplicationController`, devise will provide us with one.
- [x] [Install and setup devise](https://github.com/heartcombo/devise#getting-started).
  - [x] Can register a new user.
  - [x] User logs in with a combination of email and password.
  - [x] Hashed password should be stored in the database.
  - [x] Ask for confirmation of email.
  - [x] Can reset password.

### Not required
- [x] Refactor tests to integrate with Devise.

#### n+1
- [x] Make sure that the N+1 problem is solved when fetching all posts and their comments for a user. This is the corresponding wireframe view.

#### Integration specs
-  [x] Use Capybara to write integration tests for each view in your project.
  
- User index page:
  - [x] I can see the username of all other users.
  - [x] I can see the profile picture for each user.
  - [x] I can see the number of posts each user has written.
  - [x] When I click on a user, I am redirected to that user's show page.
  
- user show page:
  - [x] I can see the user's profile picture.
  - [x] I can see the user's username.
  - [x] I can see the number of posts the user has written.
  - [x] I can see the user's bio.
  - [x] I can see the user's first 3 posts.
  - [x] I can see a button that lets me view all of a user's posts.
  - [x] When I click a user's post, it redirects me to that post's show page.
  - [x] When I click to see all posts, it redirects me to the user's post's index page.

- User post index page:
  - [x] I can see the user's profile picture.
  - [x] I can see the user's username.
  - [x] I can see the number of posts the user has written.
  - [x] I can see a post's title.
  - [x] I can see some of the post's body.
  - [x] I can see the first comments on a post.
  - [x] I can see how many comments a post has.
  - [x] I can see how many likes a post has.
  - [x] I can see a section for pagination if there are more posts than fit on the view.
  - [x] When I click on a post, it redirects me to that post's show page.

- Post show page:
  - [x] I can see the post's title.
  - [x] I can see who wrote the post.
  - [x] I can see how many comments it has.
  - [x] I can see how many likes it has.
  - [x] I can see the post body.
  - [x] I can see the username of each commentor.
  - [x] I can see the comment each commentor left.

### CODE REVIEW

File: app/views/devise/sessions/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      <%= f.label :remember_me %>
+    </div>
+  <% end %>
+
+  <div class="actions">
+    <%= f.submit "Log in" %>
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
```
Comment (CollinsTatang) [inline app/views/devise/sessions/new.html.erb]: - [ ] Please modified the `login` to match the wireframes and their styling.
login

**Review by CollinsTatang — CHANGES_REQUESTED** [review]
Hi @rbanos-mv, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

## Required changes 
It's ok to have bugs in your code. Please, see this code review as a chance for you to learn practical coding skills and learn new tips that can improve your code quality and help you get a better job.
_Check the comments under the review._
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/devise/sessions/new.html.erb (hunk 1 above)
Comment (rbanos-mv) [reply app/views/devise/sessions/new.html.erb]: Hello @CollinsTatang .

The wireframe does not include both the "Forgot your password?" and the "Sign up" link.
Does that mean the user is expected to know/guess the URL for those required functionalities?

Please confirm what should I do to those links. Should I ommit them?

Comment (CollinsTatang) [reply app/views/devise/sessions/new.html.erb]: Hi @rbanos-mv 
As per project requirement, you are expected to follow the [Wireframe Design](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md)

Comment (CollinsTatang) [reply app/views/devise/sessions/new.html.erb]: You can check the last point on your [ project requirement ](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_devise.md)

**Review by gemmen29 — APPROVED** [review]
Hi @rbanos-mv, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 



## Optional suggestions

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**